### DISCUSSION TITLE
How to add a default effect handler?

### DISCUSSION THREAD

**Discussion by woojamon** [koka-lang/koka#463] [General]
I have a public parameterless function and I see "run debug | optimized" above it, but the function has a custom effect on it.
How can I write a "default" effect handler so that koka has a way to handle my function, without me explicitly handling the effect in the function body?

**Comment by TimWhiting** [comment]
For now, there isn't a way to add default handlers (except in `core`) - special cased in the compiler. 

Hint: create a function `defaultYourEff(action: () -> youreff a) ..` and handles it, then just add `with defaultYourEff` as the first line of the function that needs it. It's still pretty succinct.
